app/src/main/kotlin/com/wire/android/ui/calling/initiating/InitiatingCallViewModel.kt

```diff
@@ -20,28 +25,31 @@ import com.wire.kalium.logic.feature.call.usecase.ObserveEstablishedCallsUseCase
 import com.wire.kalium.logic.feature.call.usecase.StartCallUseCase
 import com.wire.kalium.logic.feature.conversation.ObserveConversationDetailsUseCase
 import dagger.hilt.android.lifecycle.HiltViewModel
-import kotlinx.coroutines.async
 import kotlinx.coroutines.flow.filterIsInstance
 import kotlinx.coroutines.flow.first
 import kotlinx.coroutines.flow.map
 import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
 import java.util.Calendar
 import javax.inject.Inject
 
 @Suppress("LongParameterList")
 @HiltViewModel
 class InitiatingCallViewModel @Inject constructor(
     savedStateHandle: SavedStateHandle,
-    private val navigationManager: NavigationManager,
     qualifiedIdMapper: QualifiedIdMapper,
+    private val navigationManager: NavigationManager,
     private val conversationDetails: ObserveConversationDetailsUseCase,
     private val observeEstablishedCalls: ObserveEstablishedCallsUseCase,
     private val startCall: StartCallUseCase,
     private val endCall: EndCallUseCase,
     private val isLastCallClosed: IsLastCallClosedUseCase,
-    private val callRinger: CallRinger
+    private val callRinger: CallRinger,
+    private val dispatchers: DispatcherProvider
```

Comment:
I think it's better to change the dispatcher in the suspend function it self and not in the viewModel(eg. usecase). This way we don't have to worry about what Dispatcher to use for what type of work.

**This responsibility lies in the class that does the work.**

Reply:
Hmmm not sure if I follow. I do use the dispatchers to change the context on the corresponding method inside the view model. 

If you mean that the suspend function calling the view model should hold the responsibility (`InitiateCallScreen` for instance) you are implying that the view layer knows already the kind of logic that is being performed on the VM (whether if it is a IO operation or not) which I'm not sure it is the case.

Reply:
I mean we don't need to change dispatcher here, we should do it in the suspend function itself(usecase for example)
All that we need to do here is starting a new coroutine and call the functions

Reply:
addressed